### PR TITLE
Refactorings in preparation of .NET 9.0 support

### PR DESCRIPTION
**Pull request by sebastienros**
Changes that would have to be made when supporting .NET 9 and can/should come in 2.0 release instead

- Formatting 
- Breaking changes on TrimEnd

This way the .NET 9 branch is only about .NET 9 changes

### CODE REVIEW

File: src/docs/releases/2.0.0.md [hunk 1]
```diff
@@ -739,3 +739,9 @@ Many type commonly used by modules can be `sealed`, which improves runtime perfo
 ### Workflow Trimming
 
 The Workflows module now has a `Trimming` feature to automatically clean up old workflow instances. See [the corresponding documentation](../reference/modules/Workflows/README.md#trimming) for details.
+
+### Obsoleting `TrimEnd`
+
+The methods `public static string TrimEnd(this string value, string trim = "")` from `OrchardCore.Mvc.Utilities` and `OrchardCore.ContentManagement.Utilities` are being obsoleted and replaced by 
+`OrchardCore.ContentManagement.Utilities.TrimEndString(this string value, string suffix)`. This was done to prepare teh code base for the next .NET 9.0 release which has a conflicting method
```
Comment (gvkries) [inline src/docs/releases/2.0.0.md:746]:
```suggestion
`OrchardCore.ContentManagement.Utilities.TrimEndString(this string value, string suffix)`. This was done to prepare the code base for the next .NET 9.0 release which has a conflicting method 
```

**Review by gvkries — APPROVED** [review]

**Review by Piedone — APPROVED** [review]

File: src/OrchardCore/OrchardCore.ContentManagement.Abstractions/Utilities/StringExtensions.cs [hunk 2]
```diff
@@ -356,6 +358,19 @@ public static string TrimEnd(this string value, string trim = "")
             ? value[..^trim.Length]
             : value;
     }
+#endif
+
+    public static string TrimEndString(this string value, string suffix)
```
Comment (hishamco) [inline src/OrchardCore/OrchardCore.ContentManagement.Abstractions/Utilities/StringExtensions.cs:363]: Could you please elaborate what are the different behaviors?

In both cases, we can have the same name and move #if .. #elseif .. #endif inside the function, this way no need to have a new method name

Comment (sebastienros) [reply src/OrchardCore/OrchardCore.ContentManagement.Abstractions/Utilities/StringExtensions.cs:363]: `TrimEnd` exists in net9.0, so we need to hide it when we build in net9.0.
And in net9.0 `TrimEnd("Part")` is equivalent to `TrimEnd(new char[] { 'P', 'a', 'r', 't' })` so `"MenuListPart"` would become `"MenuLis"` without the `t`.

Comment (hishamco) [reply src/OrchardCore/OrchardCore.ContentManagement.Abstractions/Utilities/StringExtensions.cs:363]: Seems a new overload has been added in .NET 9.0

**Pull request merged**